### PR TITLE
Add variable mapping of psm3

### PR DESCRIPTION
**Pull request by AdamRJensen**
- ~~[ ] Closes #xxxx~~
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~~[ ] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Add `map_variables` argument to `get_psm3` and `read_psm3` functions to be consistent with the general iotools function pattern.

At the moment `map_variables` defaults to None (False) and a deprecation warning has been added. The PR is similar to what was done for `get_pvgis_tmy` in #1268.

Additionally, I have implemented it such that pvlib names can be used when specifying the `attributes` arguments (requested columns), using the variable map in reverse.

### CODE REVIEW

**Comment by AdamRJensen** [comment]
It seems the `zenith` column returned by psm3 is "apparent_zenith", see comparison below:

image

However, should the psm zenith column map then be mapped to 'apparent_zenith' or 'apparent_solar_zenith'?

The pvliv [Variables and Symbols List](https://pvlib-python.readthedocs.io/en/stable/variables_style_rules.html) has 'solar_zenith' as the standard name, which is also what zenith is mapped to in the [get_cams()](https://github.com/pvlib/pvlib-python/blob/master/pvlib/iotools/sodapro.py#L18-L29) function.

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:41]:
```suggestion
    'Elevation': 'elevation',
    'Surface Albedo': 'albedo',
    'Precipitable Water': 'precipitable_water',
```

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:204]:
```suggestion
def parse_psm3(fbuf, map_variables=None):
```

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:333]: We should add a test that these warnings are emitted correctly

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:165]: This is interesting.  Do we do "reverse mapping" in any other iotools functions?  It is unfortunate that the PSM3 API's input parameter names are different from the output column names.

Maybe clearer for the second line, up to you: `attributes = [amap.get(a, a) for a in attributes]`

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:87]:
```suggestion
    map_variables: boolean, optional
        When true, renames columns of the Dataframe to pvlib variable names
        where applicable. See variable :const:`PSM3_VARIABLE_MAP`.
```

Comment (kanderso-nrel) [reply pvlib/iotools/psm3.py:87]: And the same for the other docstrings

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:344]: If these are handled separately anyway, is there any point in listing them in the mapping dict?

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [reply pvlib/iotools/psm3.py:165]: If we keep this, it should probably be mentioned in the docstring

**Review by kanderso-nrel — COMMENTED** [review]
Can you please also add `2020`, `tmy-2020`, `tdy-2020`, and `tgy-2020` to the docstring tables? :)

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py:344]: No, I suppose not.

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py:165]: I cannot immediately think of any other function where reverse mapping would make sense. Pretty nifty though! And I think it is in line with the spirit of the iotools, in that it conforms the data access interface with pvlib conventions.

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py:333]: Done!

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [reply pvlib/iotools/psm3.py:165]: @AdamRJensen should we add a sentence to the `attributes` docstring description for this?  Something like (feel free to edit):
> Alternatively, pvlib names may also be used (e.g. ``'ghi'`` rather than ``'GHI'``); see :const:`PSM3_VARIABLE_MAP`.

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:25]: Shall we change this to just `VARIABLE_MAP`?  https://github.com/pvlib/pvlib-python/pull/1368#discussion_r782620946

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [reply pvlib/iotools/psm3.py:165]: If we're referencing the iotools variable maps via `:const:` then we should think about giving them their own entries in the docs.  Something like what we did for `pvlib.temperature.TEMPERATURE_MODEL_PARAMETERS` maybe?  Let's do that in a separate issue though.

**Comment by mikofski** [comment]
@AdamRJensen  sorry for bombing this issue with irrelevant request, but would you be interested in showing your work on IO tools at SciPy conference? https://www.scipy2022.scipy.org/

**Comment by AdamRJensen** [comment]
This is just a test to see if the maintainer tags work

@pvlib/pvlib-maintainer
@pvlib/maintainer
@kanderso-nrel

**Review by kanderso-nrel — APPROVED** [review]
Thanks @AdamRJensen!  I am personally excited to not have to remember anymore that `get_psm3` returns `GHI` instead of `ghi` :)

File: pvlib/iotools/psm3.py
Comment (kanderso-nrel) [inline pvlib/iotools/psm3.py:32]: Oh, maybe we should hold off merging for a moment -- https://github.com/pvlib/pvlib-python/pull/717#issuecomment-1043365592

Comment (AdamRJensen) [reply pvlib/iotools/psm3.py]: Since a decision has been made, I have renamed `apparent_zenith` to `solar_zenith`.

**Review by wholmgren — COMMENTED** [review]
Ok with me to merge pending resolution of altitude/elevation comment below.

File: pvlib/iotools/psm3.py
Comment (AdamRJensen) [reply pvlib/iotools/psm3.py]: Interesting... I'll make a follow-up issue requesting that this term be added to the [Variables and Symbols list](https://pvlib-python.readthedocs.io/en/latest/user_guide/variables_style_rules.html#variables-style-rules).

As it is right now, it would be hard for users to figure out that `map_variables` return altitude instead of elevation. Is there a good way to denote that? E.g., adding it to the ``VARIABLE_MAP``?

File: pvlib/tests/iotools/test_psm3.py
Comment (kanderso-nrel) [inline pvlib/tests/iotools/test_psm3.py:173]: @AdamRJensen this test is failing locally for me because of an `apparent_zenith`/`solar_zenith` difference.  Shouldn't this be `solar_zenith`, and shouldn't this test have failed before we merged the PR?  I'm very confused why this wasn't failing before.

Comment (AdamRJensen) [reply pvlib/tests/iotools/test_psm3.py:173]: Yes indeed. Sorry for the slow response. Thanks for the fix!

**Pull request merged**
